### PR TITLE
[camera_android]: Fixes crash when record video on Android versions lower than 12

### PR DESCRIPTION
**Pull request by raulsvilar**
The the solution of the issue flutter/flutter#109769 in the #7073 causes a crashes when the Android SDK version is lower than 31, due to EncoderProfiles not being found on these devices. This issue was solved putting the EncoderProfiles declaration after the check for SDK version.

based on https://github.com/flutter/plugins/pull/7073 by @camsim99.

Fixes https://github.com/flutter/flutter/issues/109769 in Android lower than 12.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by camsim99 — APPROVED** [review]
Thanks for the fix! Should just need merge conflicts fixed and a second review.

**Review by bparrishMines — APPROVED** [review]
LGTM

**Pull request merged**
